sparse/tpls/KokkosSparse_spmv_mv_tpl_spec_decl.hpp

```diff
@@ -63,9 +67,14 @@ inline cudaDataType compute_type<Kokkos::Experimental::half_t>() {
 */
 template <typename ViewType, std::enable_if_t<ViewType::rank == 2, bool> = true>
 cusparseDnMatDescr_t make_cusparse_dn_mat_descr_t(ViewType &view) {
-  const int64_t rows = view.extent(0);
-  const int64_t cols = view.extent(1);
-  const int64_t ld   = view.extent(0);
+  // If the view is LayoutRight, we still need to create descr as column-major
+  // but it should be an implicit transpose, meaning dimensions and strides are
+  // swapped
+  bool transpose =
+      std::is_same_v<typename ViewType::array_layout, Kokkos::LayoutRight>;
+  const int64_t rows = transpose ? view.extent(1) : view.extent(0);
+  const int64_t cols = transpose ? view.extent(0) : view.extent(1);
+  const int64_t ld   = transpose ? view.stride(0) : view.stride(1);
```

Comment:
For stride I don't know if there is a potential unsigned/signed issue?

Reply:
@lucbv rows/cols/ld get passed into cusparseCreateDnMat below, which has int64_t as the argument type. I just made it so they are initially size_t (so no conversion on lines 75-77) and then are static_cast to int64_t when passed to that function, so now there will definitely be no conversion warnings.